includes/class-wc-payments.php

```diff
@@ -569,6 +578,10 @@ public static function register_gateway( $gateways ) {
 			}
 
 			return array_merge( $gateways, $all_upe_gateways );
+		} elseif ( WC_Payments_Features::is_upe_enabled() ) {
+			$gateways[] = self::$card_gateway;
```

Comment:
Mind that the `$gateways` parameter in the `woocommerce_payment_gateways` filter contains an array of payment gateways--potentially including other non-WCPay gateways--so we will have to either do `$gateways[] = self::$card_gateway; return $gateways;` or `return array_merge( $gateways, [ self::$card_gateway ] );` to ensure that we still return all the previous and current gateways.

I've implemented the latter in bcd7d9e04cbed8cd23844dc4e294c74df1956e7a, because it's a one-liner and more consistent.